corehq/apps/data_analytics/malt_generator.py

```diff
@@ -37,14 +38,32 @@ def generate_malt(monthspans, domains=None):
     """
     domain_names = domains or Domain.get_all_names()
     for domain_name in domain_names:
+        last_submission_date = get_last_form_submission_received(domain_name)
+        last_malt_run_dates_by_month = _get_last_run_date_for_malt_by_month(domain_name, monthspans)
         for monthspan in monthspans:
-            logger.info(f"Building MALT for {domain_name} for {monthspan}")
-            all_users = get_all_user_rows(domain_name, include_inactive=False, include_docs=True)
-            for users in chunked(all_users, 1000):
-                users_by_id = {user['id']: CouchUser.wrap_correctly(user['doc']) for user in users}
-                malt_row_dicts = _get_malt_row_dicts(domain_name, monthspan, users_by_id)
-                if malt_row_dicts:
-                    _save_malt_row_dicts_to_db(malt_row_dicts)
+            last_malt_run_date = last_malt_run_dates_by_month.get(monthspan.startdate.date(), monthspan.startdate)
+            if last_submission_date and last_submission_date >= last_malt_run_date:
+                # use this date to populate last_run_date for all MALTRows with this domain and month
+                run_date = datetime.datetime.utcnow()
+                logger.info(f"Building MALT for {domain_name} for {monthspan} up to {run_date}")
+                all_users = get_all_user_rows(domain_name, include_inactive=False, include_docs=True)
+                for users in chunked(all_users, 1000):
+                    users_by_id = {user['id']: CouchUser.wrap_correctly(user['doc']) for user in users}
+                    malt_row_dicts = _get_malt_row_dicts(domain_name, monthspan, users_by_id, run_date)
+                    if malt_row_dicts:
+                        _save_malt_row_dicts_to_db(malt_row_dicts)
+
+
+def _get_last_run_date_for_malt_by_month(domain_name, monthspans):
+    """
+    Checks for existing MALTRows for this domain and monthspan
+    If a row does not exist, or the last_run_date on that row is None, return the start date of the month
```

Comment:
This comment feels misleading since the logic that implements the described behavior is outside the fucntion (in line 44). The function actually returns nothing in that case (the row itself is excluded).

Reply:
Good catch. Meant to update that when making the recently requested change. See f833ab27f46b07679568375351f583f59fca49bd